Enable plugins from asset library by default

Jummit opened the pull request:
fixes godotengine/godot-proposals#324 and #16492

Jummit committed:
enable plugins from asset library by default

Calinou reviewed (commented):
Maybe there should be a few comments to describe why add-ons are enabled automatically :slightly_smiling_face:

Jummit commented:
Such as? I thought it was pretty standard behavior that installed addons will be enabled by default.

bojidar-bg commented:
What will happen if there are multiple addons packed as one asset?

Jummit commented:
> What will happen if there are multiple addons packed as one asset?

I didn't think of that, because that's pretty uncommon, but I'll implement it. (It currently only enables one)

Jummit commented:
Updated with version that supports multiple plugins.

_I am very frustrated because of git.
I'm going to take a walk.
Everything will be ok._

bojidar-bg commented on editor/editor_asset_installer.cpp:
```diff
@@ -261,6 +262,14 @@ void EditorAssetInstaller::ok_pressed() {
+				if (dirpath.begins_with("res://addons")) {
+					Vector<String> separated = dirpath.split("/");
+					String plugin_folder = separated[separated.find("addons") + 1];
+					if (plugin_folders.find(plugin_folder) == -1) {
```
This is `O(n^2)`, but might be improved to `O(n log n)` by using `Set<String>` instead of `Vector<String>`. Probably even faster with `Set<StringName>`, as comparisons will be faster this way.

Jummit replied:
I used a Set<StringName> instead of a Vector<String> now.

bojidar-bg replied:
No, I did not mean the `separated` variable; indeed, changing it into a Set would make it impossible to guess the plugin name.

Jummit commented:
I added a "Install and enable" button because there were people who didn't want to have it be default:
https://github.com/godotengine/godot/issues/16492#issuecomment-365168283
Also fixed some bugs.

aaronfranke commented:
Perhaps #36713 should be implemented first? Otherwise you would download an asset and immediately get errors.

Jummit commented:
Closing as there is no way this can be merged into 4.0.
